### PR TITLE
only load collections once

### PR DESCRIPTION
**Pull request by abrookins**
Any time we look up a block from a block document, we load all collections from disk, even if we've already loaded them. It seems unlikely that these will change while the process is running or that we need to support collections changing out from under a running process, so this PR loads them only once. A micro-optimization? Maybe.

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] If this pull request adds new functionality, it includes unit tests that cover the changes
- [ ] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [ ] If this pull request adds functions or classes, it includes helpful docstrings.

### CODE REVIEW

File: src/prefect/blocks/core.py [hunk 1]
```diff
@@ -72,6 +72,9 @@
 ResourceTuple = Tuple[Dict[str, Any], List[Dict[str, Any]]]
 
 
+_collections_loaded = False
```
Comment (cicdw) [inline src/prefect/blocks/core.py:75]: just as a matter of style consistency, could you make this all upper case?

**Review by cicdw — CHANGES_REQUESTED** [review]
just a style nit, otherwise love it! thank you

File: src/prefect/blocks/core.py (hunk 1 above)
Comment (abrookins) [reply src/prefect/blocks/core.py:75]: Totally down to use uppercase here. I would normally only use uppercase for a constant, while this is a mutable global variable. Do we normally uppercase those? If so, I can update my mental model.

Comment (cicdw) [reply src/prefect/blocks/core.py:75]: Yea I see what you're saying -- we do use the uppercase convention elsewhere though for mutable values, e.g.:
- https://github.com/PrefectHQ/prefect/blob/main/src/prefect/logging/configuration.py#L24
- https://github.com/PrefectHQ/prefect/blob/main/src/prefect/settings.py#L1859

Comment (abrookins) [reply src/prefect/blocks/core.py:75]: Makes sense to me, and I'm down to keep it consistent.

**Comment by desertaxle** [comment]
Would it make sense to put this optimization in the `load_collections` function directly? That way other consumers share the optimization.

**Comment by abrookins** [comment]
> Would it make sense to put this optimization in the `load_collections` function directly? That way other consumers share the optimization.

It might! We'd need to cache the output of the function and return that instead. I'll throw up a version like that for you guys to check out. Nothing crazy, but also that object will live in memory, of course.

**Comment by abrookins** [comment]
Delayed a bit while I add a test suite for the plugins module but should be ready this morning PT.

**Comment by abrookins** [comment]
I pushed up some commits with the changes we discussed. However, GitHub must be "having a moment" because while the commits appear in the branch, this PR isn't reflecting them. 🙄 A problem for tomorrow.

**Review by desertaxle — APPROVED** [review]

**Pull request merged**